### PR TITLE
Compatibility with pymanopt and solved some bugs

### PR DESCRIPTION
**Pull request by ZXP-S-works**
1. Updated _numerical.py so it becomes compatible with pymanopt.

2. I also added an octahedral conv3d test code to ./examples.

3. Lastly, I fixed some bugs for 3d convolution (padding issue and assertion issue in pointwise_adaptive_max.py, pointwise_max.py, r2upsampling.py).

Signed-off-by: ZXP-S-Works <zxpzzz521@gmail.com>

### CODE REVIEW

File: examples/octahedral cnn.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import torch
```
Comment (psteinb) [inline examples/octahedral cnn.py:1]: There is a whitespace in the filename. It would be great if you remove it or replace it by an underscore.

**Comment by Gabri95** [comment]
I am merging your changes to _numerical.py which fix the pymanopt issue.

I will look at the other changes afterwards (I just want to keep the different fixes and contributions separated and organised)

Thanks again!
Gabriele

**Comment by ZXP-S-works** [comment]
Hi Gari95 and Psteinb,

I am glad to contribute to this important repo!

In terms of the pull request, take your time to review it.

In terms of [examples/octahedral cnn.py](https://github.com/QUVA-Lab/escnn/pull/33/files/8b15f8635d6132b217d3e493371ac0035713c712#diff-8b6351e2c01b2fe489cc3a86fe8684c0f47b37da506f028f1de8112d9ba83496), I have updated the file name to "octahedral_cnn.py". Thank you for reminding me!

Best,
XP

**Comment by Gabri95** [comment]
I should have already included all the proposed changes but I forgot to close this pull-request in the merge commit, so I will close it manually now.

For reference,  [this commit](https://github.com/QUVA-Lab/escnn/tree/035f1ec29e6f9f8cf4ff17ebda3f8332e78df4ce) was the last commit implementing these changes

I've also included the 3D version of most of the other downsampling methods, see [Pooling](https://quva-lab.github.io/escnn/api/escnn.nn.html#pooling) and let me know if something is missing!

Unfortunately, I  don't have access to my pypi account at the moment so I can't push the latest version there. I should be able to do so in a few days, thanks for your patience!

Thanks again for your contribution!
Gabriele
